core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -350,6 +354,224 @@ public void testPartitionsTableScanNotNullFilter() {
     validateIncludesPartitionScan(tasksUnary, 3);
   }
 
+  @Test
+  public void testFilesTableScanNoFilter() {
+    table.newFastAppend()
+        .appendFile(FILE_PARTITION_0)
+        .commit();
+    table.newFastAppend()
+        .appendFile(FILE_PARTITION_1)
+        .commit();
+    table.newFastAppend()
+        .appendFile(FILE_PARTITION_2)
+        .commit();
+    table.newFastAppend()
+        .appendFile(FILE_PARTITION_3)
+        .commit();
+
+    Table dataFilesTable = new DataFilesTable(table.ops(), table);
+    Types.StructType expected = new Schema(
+        required(102, "partition", Types.StructType.of(
+            optional(1000, "data_bucket", Types.IntegerType.get())),
+            "Partition data tuple, schema based on the partition spec")).asStruct();
+
+    TableScan scanNoFilter = dataFilesTable.newScan().select("partition.data_bucket");
+    Assert.assertEquals(expected, scanNoFilter.schema().asStruct());
+    CloseableIterable<CombinedScanTask> tasksNoFilter = scanNoFilter.planTasks();
```

Comment:
I think this would be a bit simpler if you used `planFiles` instead of `planTasks`. That should produce only `FileScanTask` rather than `CombinedScanTask` so you wouldn't need to `flatMap` tasks because there is only one data file per task.

Reply:
I had an issue early on where planTasks caught it and planFiles did not for some reason (passing wrong schema).  I changed most of them to planFiles, and added one specific test for planTasks.